### PR TITLE
Allow translated URL patterns in path(), re_path()

### PR DESCRIPTION
**Pull request by imankulov**
URL patterns can be marked translatable using the gettext_lazy() function.

Ref: https://docs.djangoproject.com/en/4.2/topics/i18n/translation/#translating-url-patterns

### CODE REVIEW

**Comment by imankulov** [comment]
👋 I would appreciate it if someone could review this PR. My current workaround is using `cast(str, _("admin/"))`, which is acceptable but unnecessarily verbose. Thanks!

File: tests/typecheck/urls/test_conf.yml [hunk 1]
```diff
@@ -17,6 +17,17 @@
...
 
+-   case: test_path_accepts_translated_patterns
+    main: |
+      from typing import List
+      from django.urls import path, _AnyURL
+      from django.utils.translation import gettext_lazy as _
+      from django.contrib import admin
+
+      urlpatterns: List[_AnyURL] = [
+          path(_("admin/"), admin.site.urls, name="admin"),
```
Comment (intgr) [inline tests/typecheck/urls/test_conf.yml:30]: If you already added a test, let's add a test case for `re_path()` too.

Comment (intgr) [reply tests/typecheck/urls/test_conf.yml:30]: Just add another line to this test case.

**Review by intgr — APPROVED** [review]
Lookg good to me, small tweak to tests and then this is ready.

File: tests/typecheck/urls/test_conf.yml (hunk 1 above)
Comment (imankulov) [reply tests/typecheck/urls/test_conf.yml:30]: Thanks for the review, @intgr! Addressed in 0546071824520a0af35d869b2644fc01c8ce09a0

**Review by intgr — APPROVED** [review]

**Pull request merged**